apps/vmq_plugin/src/vmq_plugin_util.erl

```diff
@@ -0,0 +1,129 @@
+-module(vmq_plugin_util).
+
+-export([check_modifiers/2]).
+
+check_modifiers(auth_on_subscribe, Modifiers) ->
+    %% auth_on_subscribe take a special form
+    lists:foldl(fun (_, error) -> error;
+                    ([T, Q], Acc) when is_binary(T) and is_number(Q) ->
+                        case vmq_topic:validate_topic(subscribe, T) of
+                            {ok, Topic} ->
+                                [{Topic, to_internal_qos(round(Q))}|Acc];
```

Comment:
Yeah, also thought of that - the only other place to put it at the moment would be in vmq_commons or vernemq_dev. But then on the other hand it's kind of strongly related to plugins...